shared/src/types/storage.rs

```diff
@@ -223,6 +223,15 @@ impl Key {
         addresses
     }
 
+    /// Return the segment at the index parameter
+    pub fn get_at(&self, index: usize) -> Option<DbKeySeg> {
+        if self.segments.len() > index {
+            Some(self.segments[index].clone())
+        } else {
+            None
+        }
+    }
```

Comment:
```suggestion
    pub fn get_at(&self, index: usize) -> Option<&DbKeySeg> {
        self.segments.get(index)
    }
```
(can be chained with `.cloned()` if needed)